woocommerce/woocommerce
# Setup Wizard: Default 100% of users to new OBW.

### All Submissions:






### Changes proposed in this Pull Request:

This branch seeks to move 100% of users through the new OBW shipped inside of woocommerce-admin. Given some of the discussion in #26010 around how the current e2e test suite relies upon the legacy OBW, as does the ecommerce plan, I've taken the following approach here:

- Keep the legacy OBW code intact, and usable by setting a filter of:

`add_filter( 'woocommerce_setup_wizard_force_legacy', '__return_true' );`

- Once the e2e tests are updated to the new OBW, and a solution is in place in the ecomm plan, we can do a follow-up PR to remove the legacy OBW code completely.
- I've also changed the text at the bottom of the OBW splash screen to show a "Not Right Now" link so users can exit the OBW still.

@juliaamosova - I wasn't certain where to add the filter to force the legacy OBW for the e2e test suite.

__Before__
<img width="819" alt="WooCommerce › Setup Wizard - Mozilla Firefox 2020-03-25 13-19-28" src="https://user-images.githubusercontent.com/22080/77583575-901e1480-6e9e-11ea-9918-cd2c4be11df4.png">

__After__
<img width="819" alt="WooCommerce › Setup Wizard - Mozilla Firefox 2020-03-25 13-19-59" src="https://user-images.githubusercontent.com/22080/77583586-97452280-6e9e-11ea-8f31-90d82b8c72c6.png">

Closes #26010 
Closes #26011

### How to test the changes in this Pull Request:

1. Visit `wp-admin/admin.php?page=wc-setup` and verify the _After_ screen above appears
2. Click Yes Please and verify the new OBW is shown
3. Use the filter above and verify the old OBW is still shown.

### Changelog entry

> Enhancement - Use new Setup Wizard for all users.
